### PR TITLE
Make ivy.sparse_array inherit from ivy.array

### PR DESCRIPTION
**Pull request by vaithak**

### CODE REVIEW

**Comment by vaithak** [comment]
@vedpatwardhan, The test fails because of invalid initialization of sparse array as the `other` parameter of the `__add__` function.  

Can you suggest how I can pass the `indices, value, ...` params to properly initialize the sparse array in `handle_test_method` and use it as the `other` argument for `Array.__add__`?

**Comment by vaithak** [comment]
btw, I tested it locally, the addition of `ivy.array` and `ivy.sparse_array` works after this change

File: ivy/data_classes/array/array.py [hunk 1]
```diff
@@ -144,6 +144,10 @@ def _init(self, data, dynamic_backend=None):
             self._data = data
         elif isinstance(data, np.ndarray):
             self._data = ivy.asarray(data)._data
+        elif ivy.experimental.is_ivy_sparse_array(data):
```
Comment (vedpatwardhan) [inline ivy/data_classes/array/array.py:147]: let's use `ivy.is_ivy_sparse_array` instead, same applies to the next `if`

**Review by vedpatwardhan — CHANGES_REQUESTED** [review]
Hey @vaithak, thanks for diving into this, could you please also pull the updated code from `main` into your branch? There were some issues with the compiler tests which were failing all tests in PRs which have now been fixed, so pulling from `main` should give a better idea of the test failures specific to your changes 😄

File: ivy/data_classes/array/array.py (hunk 1 above)
Comment (vaithak) [reply ivy/data_classes/array/array.py]: updated 👍🏼

File: ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_sparse_array.py [hunk 2]
```diff
@@ -365,3 +365,51 @@ def test_sparse_bsr(
...
+        backend_to_test=backend_fw,
+        on_device=on_device,
+        ground_truth_backend=ground_truth_backend,
+        init_flags=init_flags,
+        method_flags=method_flags,
+        init_all_as_kwargs_np={"data": x[0]},
+        init_input_dtypes=dtype,
+        method_input_dtypes=["int64", val_dtype],
+        method_all_as_kwargs_np={
+            "other": {
```
Comment (vaithak) [inline ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_sparse_array.py:406]: my query is regarding this part, I want the `other` parameter to be initialised with a sparse array instance, which itself should be initialised using the parameters created (like `coo_indices`, `values`, `dense_shape`...)

Comment (vaithak) [reply ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_sparse_array.py:406]: @vedpatwardhan Any suggestions for this?

Comment (vaithak) [reply ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_sparse_array.py]: removed `test_method` and added a manual check, sort of hacky but works for now

**Comment by vaithak** [comment]
The failing tests are unrelated to the changes in this PR, they are related to tensorized operations.

**Comment by vedpatwardhan** [comment]
Yep, I think it's good to go for now. Could you please fix the merge conflicts and update your branch with the code on `main`? Thanks @vaithak 😄

**Comment by vaithak** [comment]
> Yep, I think it's good to go for now. Could you please fix the merge conflicts and update your branch with the code on `main`? Thanks @vaithak 😄

updated with main 👍🏼

**Comment by vedpatwardhan** [comment]
awesome, feel free to merge the PR, thanks @vaithak 😄

**Pull request merged**